gsa/src/web/graphql/scanconfigs.js

```diff
@@ -448,3 +510,213 @@ export const useDeleteScanConfigsByIds = options => {
+    options,
+  );
+  const [queryModifyScanConfigSetFamilySelection] = useMutation(
+    MODIFY_SCAN_CONFIG_SET_FAMILY_SELECTION,
+    options,
+  );
+
+  const modifyScanConfig = useCallback(
+    // eslint-disable-next-line no-shadow
+    (saveData, options) => {
+      const {name, id} = saveData;
+      return queryModifyScanConfigSetName({
+        ...options,
+        variables: {
+          input: {
+            name,
+            id,
+          },
+        },
+      }).then(() => {
+        const {comment} = saveData;
+
+        return queryModifyScanConfigSetComment({
+          ...options,
+          variables: {
+            input: {
+              id,
+              comment,
+            },
+          },
+        }).then(() => {
+          const {scannerPreferenceValues} = saveData;
+
+          let setScannerPreferencePromise;
+
+          if (hasValue(scannerPreferenceValues)) {
+            const prefKeys = Object.keys(scannerPreferenceValues);
+
+            const prefPromises = [];
+
+            prefKeys.forEach(key => {
+              prefPromises.push(
+                queryModifyScanConfigSetScannerPreference({
+                  ...options,
+                  variables: {
+                    input: {
+                      id,
+                      name: 'scanner:scanner:scanner:' + key,
+                      value: scannerPreferenceValues[key],
+                    },
+                  },
+                }),
+              );
+            });
+            setScannerPreferencePromise = Promise.all(prefPromises);
+          } else {
+            setScannerPreferencePromise = Promise.resolve();
+          }
+
+          return setScannerPreferencePromise.then(() => {
+            const {select, trend} = saveData;
+
+            let setConfigFamilySelectionPromise;
+            if (hasValue(select) && hasValue(trend)) {
+              const families = [];
+              const familyKeys = Object.keys(select);
+
+              familyKeys.forEach(key => {
+                families.push({
+                  name: key,
+                  growing: trend[key],
+                  all: select[key],
+                });
+              });
+
+              setConfigFamilySelectionPromise = queryModifyScanConfigSetFamilySelection(
+                {
+                  ...options,
+                  variables: {
+                    input: {
+                      id,
+                      families,
+                    },
+                  },
+                },
+              );
+            } else {
+              setConfigFamilySelectionPromise = Promise.resolve();
+            }
+
+            return setConfigFamilySelectionPromise;
+          });
+        });
+      });
+    },
+    [
+      queryModifyScanConfigSetName,
+      queryModifyScanConfigSetComment,
+      queryModifyScanConfigSetScannerPreference,
+      queryModifyScanConfigSetFamilySelection,
+    ],
+  );
+
+  return modifyScanConfig;
+};
```

Comment:
If name, comment, preferences and family selection are set only in one go the backend api should be changed to have only one mutation to change them at the same time. That would make the frontend api easier to implement and to read. Maybe it was also a bad idea to split that in python-gvm.

Reply:
We could recreate a single call in `python-gvm` and integrate it into the backend easily i guess ...

Reply:
> If name, comment, preferences and family selection are set only in one go the backend api should be changed to have only one mutation to change them at the same time. That would make the frontend api easier to implement and to read. Maybe it was also a bad idea to split that in python-gvm.

And also, currently scanner preferences can only be sent one at a time so I'm pretty sure pressing "save scan config" necessitates 20+ graphql calls. That is quite suboptimal. Although in my experience, sending one huge command (or requesting a list of scan configs with a lot of fields) also causes timeout sometimes.